### ISSUE TITLE
Using commands feels a bit awkward at the moment

### ISSUE THREAD

**Issue by huxi** [cbeust/jcommander#54]
Using commands isn't as easy as the rest of JCommander.

I have code that looks like this:
```
    CommandLineArgs cl=new CommandLineArgs();
    JCommander commander = new JCommander(cl);
    Cat cat = new Cat();
    commander.addCommand(Cat.NAME, cat);
    Tail tail = new Tail();
    commander.addCommand(Tail.NAME, tail);
    Index index = new Index();
    commander.addCommand(Index.NAME, index);
```
The thing that's bugging me is that I had to define a public static final NAME field in each of my command instead of annotating the class, simply calling 
```
    addCommand(command)
```
The same is the case for using the command:
```
    String command = commander.getParsedCommand();
    if(Tail.NAME.equals(command)) {
        [...]
    }
```
So there are two places where I still need to juggle with plain strings.
Instead, I'd like to have something like
```
    Object commandObject = commander.getParsedCommandObject();
```
This would be really useful if all registered command objects would implement the same interface, e.g. Runnable. In that case, I could simply cast the returned commandObject and call run() (after checking against null, obviously).

This functionality could even be wrapped into the getParsedCommandObject() method like this:
```
    Runnable commandObject = commander.getParsedCommandObject<Runnable.class>();
```

**Comment by jeremysolarz** [comment]
@cbeust an old one but some nice thoughts about `Command`s in it.

What do you say?

**Comment by cbeust** [comment]
Yes, I agree, lot of good ideas here. I'll go over it and think about how to improve the design of commands, let's keep this open.

**Comment by acostalima** [comment]
In the last project I used JCommander in (not very recently), I used `Enum`s to wrap all my command classes. Each `Enum` had two fields: a `String` to hold the command name and a `Class<?>` to hold the corresponding class type of the command. This approach helped me work around the problem @huxi described with an utility like [getEnum from Apache Commons](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/EnumUtils.html#getEnum-java.lang.Class-java.lang.String-).

**Comment by mkarg** [comment]
As @cbeust thinks we had good ideas here, wouldn't it be cool if someone of us would open a PR? 8-)
